client/src/components/AddPost/BasicSelect.tsx

```diff
@@ -0,0 +1,47 @@
+import Box from '@mui/material/Box';
+import InputLabel from '@mui/material/InputLabel';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+import { ReactNode } from 'react';
+import IPost from '../../Interfaces/post/IAddPost';
+import ITag from '../../Interfaces/post/ITag';
+
+interface Props{
+    itemId:string
+    post:IPost
+    callback:Function
+    name:string
+    obj: ITag[]
+}
+
+const BasicSelect = ({
+  name, obj, post, callback, itemId,
+} :Props) => {
+  const handleChange = (event: SelectChangeEvent<ReactNode>) => {
+    if (itemId === 'TagId') {
+      callback({ ...post, TagId: event.target.value });
+    } else {
+      callback({ ...post, AnimalId: event.target.value });
+    }
+  };
```

Comment:
No need to separate this component if its going to be hardcoded

Reply:
Which can be solved by smart typing

Reply:
cc: @mohmmed23

Reply:
We only have these two options, so we did it